### PR TITLE
detect-detection: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
Ticket: #4019

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR: https://github.com/OISF/suricata/pull/8088

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4019) ticket:

Describe changes:

- Update detect-detection unit tests to FAIL/PASS APIs

### CODE REVIEW

File: src/detect-detection-filter.c [hunk 1]
```diff
@@ -409,16 +405,12 @@ static int DetectDetectionFilterTestSig1(void)
...
-    }
+    FAIL_IF_NULL(de_ctx);
 
     de_ctx->flags |= DE_QUIET;
 
-    s = de_ctx->sig_list = SigInit(de_ctx,"alert tcp any any -> any 80 (msg:\"detection_filter Test\"; detection_filter: track by_dst, count 4, seconds 60; sid:1;)");
-    if (s == NULL) {
-        goto end;
-    }
+    s = de_ctx->sig_list = DetectEngineAppendSig(de_ctx,"alert tcp any any -> any 80 (msg:\"detection_filter Test\"; detection_filter: track by_dst, count 4, seconds 60; sid:1;)");
```
Comment (victorjulien) [inline src/detect-detection-filter.c:412]: incorrect usage of `DetectEngineAppendSig`. It should not set `de_ctx-sig_list`

Comment (victorjulien) [reply src/detect-detection-filter.c:412]: make sure to update the others in the patch as well

**Comment by victorjulien** [comment]
Formatting check failed as well, can you check that?

**Comment by hsadia538** [comment]
Followed by #8108

**Pull request closed without merging**